### PR TITLE
chore(broker): remove obsolete json-path module

### LINKED ISSUE #4059: Remove unused JSON path module
**Description**

The module `zeebe-msgpack-json-path` is not used anymore since all JSON path-like expressions are migrated to FEEL expressions. Let's remove it :wastebasket:  

requires #3808
requires #3806
requires #3804

### PR DESCRIPTION
**Pull request by pihme**
## Description

removed the json-path module

## Related issues

closes #4059

## Pull Request Checklist

- [ X ] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ X ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ X ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by pihme** [comment]
Not sure who wants to do the review. Shouldn't be too taxing.

**Comment by korthout** [comment]
LGTM. I'd assume at least some tests to fail if this was still used somewhere.

**Review by korthout — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -32,7 +34,6 @@
     <module>dispatcher</module>
     <module>msgpack-core</module>
     <module>msgpack-value</module>
-    <module>json-path</module>
```
Comment (saig0) [inline pom.xml:35]: The dependency to the `json-path` module is still declared in the `bom/pom.xml` as artifact `zeebe-msgpack-json-path` ;)

**Review by saig0 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (pihme) [reply pom.xml:35]: It would have been too easy otherwise ;-)

**Review by saig0 — APPROVED** [review]
:tada: 

RIP `json-path` (5,777 loc) :wastebasket:

**Pull request merged**